src/core/Akka.Streams/Dsl/Keep.cs

```diff
@@ -55,7 +56,11 @@ public static class Keep
         /// <returns>TBD</returns>
         public static NotUsed None<TLeft, TRight>(TLeft left, TRight right) => NotUsed.Instance;
 
+#if !CORECLR
         private static readonly RuntimeMethodHandle KeepRightMethodhandle = typeof(Keep).GetMethod(nameof(Right)).MethodHandle;
+#else
+        private static readonly MethodInfo KeepRightMethodInfo = typeof(Keep).GetMethod(nameof(Right));
```

Comment:
`IsRight` and `IsLeft` are used during materialization for [performance improvements](https://github.com/akkadotnet/akka.net/blob/4805420124154c804ff4dddcc84d9a09a4817070/src/core/Akka.Streams/Implementation/StreamLayout.cs#L730) and for [composing a materialized value with `EmptyModul` ](https://github.com/akkadotnet/akka.net/blob/4805420124154c804ff4dddcc84d9a09a4817070/src/core/Akka.Streams/Implementation/StreamLayout.cs#L955) . 

For me it looks like it _should_ work, but you should be able to verify it by debugging some tests, maybe [this one](https://github.com/akkadotnet/akka.net/blob/aa913c26cfb3f92f0a6d3ad1f24e1533cc5a9277/src/core/Akka.Streams.Tests/Dsl/GraphMatValueSpec.cs#L252) and check if one of the methods return true or the exception is thrown in the [`EmptyModule`](https://github.com/akkadotnet/akka.net/blob/4805420124154c804ff4dddcc84d9a09a4817070/src/core/Akka.Streams/Implementation/StreamLayout.cs#L955)